gateways/node_gateway.py

```diff
@@ -77,10 +78,51 @@ def get_node(self, id: str) -> Union[Node, None]:
 
         return None
 
+    def save_network(self, network: Network) -> Union[bool, None]:
```

Comment:
Shouldn't we have created a `network_gateway.py` for those methods?

I think the pattern was to have one gateway for each class of the domain, right?

Reply:
Yes, but since the aggregation part will need to call or implement the `node_gateway` methods, one would import the other, so for now it seems best to keep this as a single class.
Or we could rename `node_gateway` to `network_gateway` since it interacts with the network.